src/python/requirements/requirements.go

```diff
@@ -0,0 +1,91 @@
+package requirements
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type Reqs struct{}
+
+func (m Reqs) FindAnyPackage(buildDir string, searchedPackages ...string) (bool, error) {
+	reqPackages, err := parseRequirementsWithoutVersion(filepath.Join(buildDir, "requirements.txt"))
+	if err != nil {
+		return false, err
+	}
+
+	for _, searchedPackage := range searchedPackages {
+		if containsPackage(reqPackages, searchedPackage) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (m Reqs) FindStalePackages(oldRequirementsPath, newRequirementsPath string, excludedPackages ...string) ([]string, error) {
+	var stalePackages []string
+
+	oldPkgs, err := parseRequirements(oldRequirementsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	newPkgs, err := parseRequirements(newRequirementsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, oldPkg := range oldPkgs {
+		if !containsPackage(newPkgs, oldPkg) && !packageIsExcluded(excludedPackages, oldPkg) {
+			stalePackages = append(stalePackages, oldPkg)
+		}
+	}
+
+	return stalePackages, nil
+}
+
+func containsPackage(packages []string, searchedPackage string) bool {
+	for _, pkg := range packages {
+		if pkg == searchedPackage {
+			return true
+		}
+	}
+	return false
+}
+
+func packageIsExcluded(excludedPackages []string, packageFullName string) bool {
+	regex := regexp.MustCompile(`(?m)^[\w\-\w\[\]]+`)
+
+	packageWithoutVersion := regex.FindString(packageFullName)
+
+	for _, excludedPackage := range excludedPackages {
+		if packageWithoutVersion == excludedPackage {
+			return true
+		}
+	}
+
+	return false
+}
+
+func parseRequirements(requirementsPath string) ([]string, error) {
```

Comment:
It looks like [pip-diff only outputs the names](https://github.com/heroku-python/pip-pop/blob/master/bin/pip-diff#L98), so you may need to adjust this.

Doing so might make this implementation almost identical to `parseRequirementsWithoutVersion`. If that's so, you might consider having a single implementation and pass a bool flag on whether or not to include versions

Reply:
Or perhaps a single implemetation that returns `Package` objects instead of strings, where `Package` looks like:

```go
struct Package {
    name string
    versionConstraint string
}```

then you can use the output as needed

Reply:
I got confused with these tests, thats why I change the logic to support versioning in `pip-diff` https://github.com/cloudfoundry/python-buildpack/blob/7e1bfd3c2b65543b1ecf0cc42ab1d27658e84493/src/python/supply/supply_test.go#L526

Reply:
Oh I see. You're correct. The output will have the version constraint included ([because of this line](https://github.com/heroku-python/pip-pop/blob/master/bin/pip-diff#L80)).

However, the diff-ing part must ignore the version, [as pip-diff does](https://github.com/heroku-python/pip-pop/blob/master/bin/pip-diff#L90)